src/test/java/uk/gov/companieshouse/company_appointments/repository/CompanyAppointmentRepositoryTest.java

```diff
@@ -0,0 +1,391 @@
+                SortMapper.APPOINTED_ON, null, 0, 5, false, false);
+
+        // then
+        assertEquals(2, result.size());
+        assertEquals(APPOINTMENT_ID + 2, result.get(0).getId());
+        assertEquals(APPOINTMENT_ID + 1, result.get(1).getId());
+    }
+
+    @Test
+    void shouldFindAllAppointmentsForCompanySortedBySurname() {
+        // given
+        insertAppointment(APPOINTMENT_ID + 1, SURNAME + "B", DIRECTOR.getRole(), APPOINTED_ON_BASE,
+                null);
+        insertAppointment(APPOINTMENT_ID + 2, SURNAME + "A", DIRECTOR.getRole(),
+                APPOINTED_ON_BASE.plusDays(1), null);
+        // when
+        List<CompanyAppointmentData> result = repository.getCompanyAppointmentData(COMPANY_NUMBER,
+                SortMapper.SURNAME, null, 0, 5, false, false);
+
+        // then
+        assertEquals(2, result.size());
+        assertEquals(APPOINTMENT_ID + 2, result.get(0).getId());
+        assertEquals(APPOINTMENT_ID + 1, result.get(1).getId());
+    }
+
+    @Test
+    void testFetchAppointmentForCompanyWhenStartIndexIsLargerThanSizeOfListThrowsNotFoundException() {
+        insertAppointment(APPOINTMENT_ID + 1, SURNAME + "B", SECRETARY.getRole(), APPOINTED_ON_BASE,
+                null);
+        insertAppointment(APPOINTMENT_ID + 2, SURNAME + "A", SECRETARY.getRole(),
+                APPOINTED_ON_BASE.plusDays(2), null);
+        insertAppointment(APPOINTMENT_ID + 3, SURNAME + "C", DIRECTOR.getRole(),
+                APPOINTED_ON_BASE.plusDays(1), null);
+        insertAppointment(APPOINTMENT_ID + 4, SURNAME + "A", DIRECTOR.getRole(),
+                APPOINTED_ON_BASE.plusDays(1), null);
+
+        List<CompanyAppointmentData> result = repository.getCompanyAppointmentData(COMPANY_NUMBER,
+                null, null, 4, 5, false, false);
+
+        assertTrue(result.isEmpty());
+    }
+
+    @SuppressWarnings({"unchecked"})
+    private void insertAppointment(String appointmentId, String surname, String officerRole,
+            LocalDateTime appointedOn, LocalDateTime resignedOn) {
+        templateDocument.put("appointment_id", appointmentId);
+        templateDocument.put("_id", appointmentId);
+
+        Map<String, Object> officerData = (Map<String, Object>) templateDocument.get("data");
+        officerData.put("officer_role", officerRole);
+        switch (officerRole) {
+            case "director":
+            case "corporate-director":
+            case "nominee-director":
+            case "corporate-nominee-director":
+                templateDocument.put("officer_role_sort_order", resignedOn == null ? 20 : 200);
+                break;
+            case "secretary":
+            case "corporate-secretary":
+            case "nominee-secretary":
+            case "corporate-nominee-secretary":
+                templateDocument.put("officer_role_sort_order", resignedOn == null ? 10 : 100);
+                break;
+            case "llp-member":
```

Comment:
these cases are duplicates with "director", as are "llp-designated-member" with "secretary"

Reply:
Good spot but it doesn't affect tests so I think it can stay